GraphQL.md

```diff
@@ -78,62 +93,139 @@ book(id: 1) {
 
-## Deletes
-
-A set of `DELETE` examples.
+## REMOVE 
+```
+book(id: 1) {
+    authors(op: REMOVE, id: 3)
+}
+```
+Removes the _association_ between book with `id = 1` and author with `id = 3`, however, the author is still present in the persistence store.
+## REPLACE
+A set of `REPLACE` examples.
 
-### Delete a Book
+### Replace All Book Authors
 ```
-mutation book(op: DELETE, id: 1) {
-  id
+mutation book(op: REPLACE, id: 1, data: {authors:[{ name: "The New Author" }]}) {
+  id,
+  authors
 }
 ```
 
+## Complex Queries
+### Replacing a particular nested field
+Let's assume that in a complex scenario, we want to update the name of the 18th author of the 9th book. The corresponding query would be,
+```
+book(id: 9) {
+    id,
+    authors(op: REPLACE, id: 18, data: {name: "New author"}) {
+        title
+    }
+}
+```
+The above payload structure helps us manipulate a specific entity amongst several different entities linked with to same parent as under.
+```
+book(id = 9)
+| \ \
+.. .. authors(id = 18)
+      |
+      name
+```
+### Replacing two seperate fields linked to the same parent
+Let's say we want to replace the title of two seperate books associated with the same author. The corresponding query would look like,
+```
+author(id: 1) {
+    id, 
+    books(op: REPLACE, data: [{id: 1, title: "New title"}, {id: 2, title: "New title"}]) {
+        id
+    }
+}
+```
+The above payload structure helps us manipulate attributes associated with two different entities having the same parent entity as under.
+```
+author
+|     \
+|      \
+books   books
+|  \    |  \
+|   \  ...  title
+...  title
+```
+### Replacing fields of two seperate entities associated to the same parent
+Now lets say we want to modify a ``Book`` and a `Publisher` name. This can be accomplished in a single query as under.
+```
+author(id: 1) {
+    id, 
+    books(op: REPLACE, data: [{id: 1, title: "New title"}]) {
+        id
+    }, 
+    publishers(op: REPLACE, data: [{id: 1, name: "New name"}]) {
+        id
+    }
+}
+```
+The above payload structure helps us manipulate attributes of two seperate entities associated with the same parent in a single transaction as under. 
+```
+author
+|     \ 
+books  publishers
+|      |
+title  name
+```
+### Allowing multiple operations in a single transaction
+We can get fancy and allow for multiple operations, like replacing title of a book and deleting a publisher, all in a single transaction. 
+```
+author(id: 1) {
+    id, 
+    books(op: REPLACE, data: [{id: 1, title: "New title"}]) {
+        id
+    }, 
+    publishers(op: REMOVE, id: 1) {
+        id
+    }
+}
+```
 # Semantics
 
 Below is a chart of expected behavior from GraphQL queries:
 
 | Operation | Id Parameter | Id in Body | Behavior |
 | --------- | ------------ | ---------- | -------- |
-| Add       | True         | True       |  Must match. Implies update. |
+| Upsert    | True         | True       |  Must match. Implies update. |
```

Comment:
I think we should change the creation logic to behave more like an upsert.  We first attempt to load the object and if not present, we create it.  The comments below should reflect that.